### PR TITLE
Skip Go target normalization in integration tests

### PR DESCRIPTION
**Pull request by thockin**
Followup from #127480 

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: hack/install-etcd.sh [hunk 1]
```diff
@@ -25,4 +25,6 @@ set -o pipefail
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
+FOUND="$(echo $PATH | sed 's/:/\n/g' | grep -q "^${KUBE_ROOT}/third_party/etcd$" || true)"
```
Comment (BenTheElder) [inline hack/install-etcd.sh:28]: should we check the version?

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve
/hold

File: hack/install-etcd.sh (hunk 1 above)
Comment (thockin) [reply hack/install-etcd.sh:28]: The point was only to tell users that etcd is not in their path.  We'll call the install function either way.

**Comment by pohly** [comment]
> I am stumped with this vmodule error.

Not all tests register the klog logging flags. `gci.test` apparently doesn't. Either it wasn't tested before, or `-vmodule` was not passed before. It looks like https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127606/pull-kubernetes-integration/1838727817513144320 now runs all unit tests, not just integration tests?

**Comment by thockin** [comment]
```
go: list -m cannot be used with GO111MODULE=off
```

Why are these tests running with GO111MODULE=off ?

**Comment by thockin** [comment]
OK, This PR includes a small change with huge impact - turning modules on for tests.  It seems to work, but I will break it to a new PR.

**Comment by thockin** [comment]
Rebased now that GO111MODULE pr is in.

**Review by pohly — APPROVED** [review]
/lgtm

Number of tests in pull-kubernetes-integration and pull-kubernetes-unit are the same as before.

/hold cancel

**Pull request merged**